rapids_build_backend/impls.py

```diff
@@ -51,6 +52,23 @@ def _get_backend(build_backend):
         )
 
 
+@lru_cache
+def _get_arch() -> str:
+    """Get the arch of the current machine.
+
+    Returns
+    -------
+    str
+        The arch (e.g. "x86_64" or "aarch64")
+    """
+    plat = platform.machine()
+    # RAPIDS projects all use "aarch64" to indicate arm architectures,
+    # but arm some systems (like the M1/M2/M3 Macs) report "arm64"
+    if plat == "arm64":
+        return "aarch64"
```

Comment:
I'm fine with this for now, but it does feel like a bit of a hack. Technically `platform.machine` is probably not sufficiently discriminating between types of architectures in many cases, and in this particular case it is giving us a meaningful difference that we are choosing to ignore: our aarch64 dependency lists throughout RAPIDS are probably not valid on most Macs anyway. I'm comfortable with including this given that RBB is specialized enough for RAPIDS's needs that I don't expect anyone else to ever use it, but this line feels like a bit of a footgun in case anyone ever did try to.

Reply:
I am going to leave this as-is for now. I don't anticipate this causing issues. xref: https://github.com/rapidsai/rapids-build-backend/pull/54#discussion_r1794335781